cmd/cli/lint.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+type LintResult struct {
+	Valid    bool     `json:"valid"`
+	Warnings []string `json:"warnings"`
+	Errors   []string `json:"errors"`
+}
+
+func init() {
+	lintCmd := &cobra.Command{
+		Use:   "lint",
+		Short: "Lint feature flag configuration",
+		Long:  `Validate feature flag configuration file for common issues and best practices`,
+		RunE:  runLint,
+	}
+
+	lintCmd.Flags().StringVarP(&configFile, "config", "c", "", "Path to flag configuration file (required)")
+	lintCmd.MarkFlagRequired("config")
+
+	rootCmd.AddCommand(lintCmd)
+}
+
+func runLint(cmd *cobra.Command, args []string) error {
+	config, err := loadConfig(configFile)
+	if err != nil {
+		return err
+	}
+
+	result := lintConfig(config)
+	output, err := json.MarshalIndent(result, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error formatting output: %w", err)
+	}
+
+	fmt.Println(string(output))
+	return nil
+}
+
+func lintConfig(config *FlagConfig) *LintResult {
```

Comment:
**suggestion:** You have re-implemented the lint logic that is already available in `cmd/lint/linter.go`, I would suggest to replace your logic by this one in order to stay consistent.